salto-io/salto
# SALTO-1713: SBAA Approval Rules with Custom Condition deploy

Add support for addition of **ApprovalRule** instances with **Custom Condition**.
 
---

Test Scenarios:
- I've ran a big deployment of various additions (Some Rules were set with **sbaa__ConditionsMet__c = All** and some with **Custom**) to make sure we support the old form and the dedicated group.

Three groups were deployed in that process:
1. The dedicated group I've added handling to in this PR (With ApprovalRule with Custom defined and their Approval Conditions)
2. Additions of ApprovalRule instances without **Custom** conditions Met.
3. Additions of ApprovalCondition instances.

- Modification of **ApprovalRule**
- Modification of **ApprovalCondition**
- Addition of Circular Dependencies in Product2 instances (To make sure I didn't break the previous flow)

A few notes on this PR:
- I've created a dedicated group for both **sbaa__ApprovalRule__c** and**sbaa__ApprovalCondition__c** instances addition, which is the first time we have a ChangeGroup with two different Data types.
- We run three deploys in that scenario:
1. Deploy all the **Approval Rule Instances** where we replace **sbaa__ConditionsMet__c** from **Custom** to **ALL**
2. Deploy all the **Condition Instances**
3. Deploy again (modification) with all the **Approval Rule Instances** with **Custom** conditions met.
- Whoever wants to use this feature, will have to disable [this CV](https://github.com/salto-io/salto/blob/f5b79dfb0de08b28e2892f70fe4b57d409682ce8/packages/salesforce-adapter/src/change_validators/sbaa_approval_rules_custom_condition.ts#L33). We can decide to delete it as-well.
 
---
_Release Notes_: 
Salesforce Adapter:
- Support additions of **SBAA Approval Rule** instances with **Custom** conditions met.

---
_User Notifications_: 
_None_
